tests/python/gen_xdr.py

```diff
@@ -0,0 +1,187 @@
+
+from helpers import (TX_SET_SIZE, NETWORK_NAME, MIN_FEE,
+                     load_accounts, get_sequences_multiple_endpoints)
+
+
+AVG_BLOCK_TIME = 5  # seconds
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s %(levelname)s %(message)s')
+
+
+def parse_args():
+    """Generate and parse CLI arguments."""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--length', required=True, type=int, help='Test length in seconds')
+    parser.add_argument('--txs-per-ledger', required=True, type=int, help='Transaction rate to submit (spam) in parallel for every ledger round')
+    parser.add_argument('--prioritizer-seeds-file', required=True, type=str, help='File path to prioritizer seeds file')
+    parser.add_argument('--spammer-seeds-file', required=True, type=str, help='File path to spammer seeds file')
+    parser.add_argument('--out', default='spam-results-{}.json'.format(str(int(time.time()))), type=str, help='Spam results JSON output')
+
+    parser.add_argument('--passphrase', type=str, help='Network passphrase')
+    parser.add_argument('--horizon', action='append',
+                        help='Horizon endpoint URL (use multiple --horizon flags for multiple addresses)')
+
+    return parser.parse_args()
+
+
+async def generate_spam_tx_xdrs(prioritizers, builders, length, tx_per_ledger):
+    """Generate transaction XDR objects for each builder according to given spam rate.
+
+    NOTE each spam builder generates a single XDR, thus it should be used only once during a spam test.
+    This is to avoid sequence number collissions in case transactions do not get processed,
+    which is possible for unprioritized transactions and other unknown cases (which this test attempts to uncover).
+    """
+    rounds_num = math.ceil(length // AVG_BLOCK_TIME)
+    if len(builders) != tx_per_ledger * rounds_num:
+        raise RuntimeError('Amount of spammer seeds should exactly match (transactions per ledger * spam round numbers)')
+
+    with concurrent.futures.ProcessPoolExecutor() as process_pool:
+        futurs = []
+        remaining_builders = builders
+        for rnd in range(rounds_num):
+            # pop builders for this round and trim original builder list
+            round_builders = remaining_builders[-tx_per_ledger:]
+            remaining_builders = remaining_builders[:-tx_per_ledger]
+
+            assert round_builders
+
+            coro = generate_spam_round_tx_xdrs(process_pool,
+                                               prioritizers,
+                                               round_builders[:TX_SET_SIZE - 1],
+                                               round_builders[TX_SET_SIZE-1:tx_per_ledger],
+                                               rnd)
+
+            futurs.append(asyncio.create_task(coro))
+
+        spam_rounds = await asyncio.gather(*futurs)
+
+    return spam_rounds
+
+
+def build_and_sign(builder, dest_address, payment_amount, prioritizer_seed=None):
```

Comment:
```append_payment_op``` builds the operation and then adds it to the builder. You always have the same operation so there is no need to build it thousands of times.

you can build it once, and then just append it to each builder.
builder.ops.append(my_payment_op)